File: WebViewApp/app/src/main/java/com/example/webviewapp/contract/MainContract.java
package com.example.webviewapp.contract;

import com.example.webviewapp.data.Record;

import java.util.List;

public interface MainContract {
    interface View {

    }

    interface Presenter {
        void addHistory(String url, String title);

        List<String> getHistory();

        void addLabel(String url, String title);

        List<String> getLabelUrl();

        List<Record> getHistoryRecord();
    }
}

File: WebViewApp/app/src/main/java/com/example/webviewapp/contract/MainContract.kt
package com.example.webviewapp.contract

import com.example.webviewapp.data.Record

interface MainContract {
    interface View
    interface Presenter {
        fun addHistory(url: String?, title: String?)
        val history: List<String?>?
        fun addLabel(url: String?, title: String?)
        val labelUrl: List<String?>?
        val historyRecord: List<Record?>?
        fun deleteRecord(url: String?)
    }
}